### PR TITLE
[macOS] Az powershell modules fix

### PR DESCRIPTION
**Pull request by LeonidLapshin**
# Description
Az modules installation is flaky at the moment. It uses PowershellGet module for downloading dependencies so it should be updated before actual installation

## Check list
- [ ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/powershell.sh [hunk 1]
```diff
@@ -9,6 +9,10 @@ brew cask install powershell
 # A dummy call of `az` to initialize ~/.azure directory before the modules are installed
 az -v
 
+# Update PowerShellGet to eliminate download errors
+pwsh -Command "Install-Module -Name PowerShellGet -Force"
+pwsh -Command "Update-Module -Name PowerShellGet -Force"
```
Comment (al-cheb) [inline images/macos/provision/core/powershell.sh:14]: Why do we need right after the install Update-Module?

File: images/macos/provision/core/powershell.sh [hunk 2]
```diff
@@ -9,6 +9,10 @@ brew cask install powershell
 # A dummy call of `az` to initialize ~/.azure directory before the modules are installed
 az -v
 
+# Update PowerShellGet to eliminate download errors
+pwsh -Command "Install-Module -Name PowerShellGet -Force"
```
Comment (maxim-lobanov) [inline images/macos/provision/core/powershell.sh:13]: Can we just add it to toolset?
https://github.com/actions/virtual-environments/blob/main/images/macos/toolsets/toolset-10.15.json#L113

**Review by vsafonkin — APPROVED** [review]

File: images/macos/provision/core/powershell.sh [hunk 3]
```diff
@@ -17,11 +17,11 @@ for module in ${psModules[@]}; do
     echo "Installing $module module"
     moduleVersions="$(get_toolset_value ".powershellModules[] | select(.name==\"$module\") | .versions[]?")"
     if [[ -z "$moduleVersions" ]];then
-        pwsh -command "& {Install-Module $module -Force -Scope AllUsers}"
+        pwsh -command "& {[Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::Tls12; Install-Module $module -Force -Scope AllUsers}"
```
Comment (maxim-lobanov) [inline images/macos/provision/core/powershell.sh:20]: Do we really need to invoke this command in every PowerShell session?
Can we call it once before?
```
pwsh -command "& {[Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::Tls12"
```

Comment (LeonidLapshin) [reply images/macos/provision/core/powershell.sh:20]: no, it should be invoked on each session that should use tls 1.2 for sure

Comment (al-cheb) [reply images/macos/provision/core/powershell.sh:20]: No need at all:

```
PS > ${function:Save-Module}
    Begin {
        # Change security protocol to TLS 1.2
        $script:securityProtocol = [Net.ServicePointManager]::SecurityProtocol
        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12

        Install-NuGetClientBinaries -CallerPSCmdlet $PSCmdlet -Proxy $Proxy -ProxyCredential $ProxyCredential

        # Module names already tried in the current pipeline for InputObject parameterset
        $moduleNamesInPipeline = @()
    }
```

File: images/macos/provision/core/powershell.sh [hunk 4]
```diff
@@ -17,11 +17,12 @@ for module in ${psModules[@]}; do
...
     else
         for version in ${moduleVersions[@]}; do
             echo " - $version"
-            pwsh -command "& {Install-Module $module -RequiredVersion $version -Force -Scope AllUsers}"
+            tlsCommand="[Net.ServicePointManager]::SecurityProtocol = [Net.ServicePointManager]::SecurityProtocol -bor [Net.SecurityProtocolType]::Tls12"
```
Comment (al-cheb) [inline images/macos/provision/core/powershell.sh:24]: The same as above it's already set:

```
PS > ${function:Install-Module}
   Begin {
        # Change security protocol to TLS 1.2
        $script:securityProtocol = [Net.ServicePointManager]::SecurityProtocol
        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
```

Comment (LeonidLapshin) [reply images/macos/provision/core/powershell.sh:24]: yes, true, find out that while save-module is executing, tls if forced to 1.2 and is set to default at the end of execution

**Pull request closed without merging**